--- a/ext/alsa/gstalsasink.c.ori	2014-12-15 16:46:59.000000000 +0800
+++ b/ext/alsa/gstalsasink.c	2016-03-30 09:58:33.396338564 +0800
@@ -817,7 +817,7 @@
   alsa->channels = GST_AUDIO_INFO_CHANNELS (&spec->info);
   alsa->buffer_time = spec->buffer_time;
   alsa->period_time = spec->latency_time;
-  alsa->access = SND_PCM_ACCESS_RW_INTERLEAVED;
+  alsa->access = SND_PCM_ACCESS_MMAP_INTERLEAVED;
 
   if (spec->type == GST_AUDIO_RING_BUFFER_FORMAT_TYPE_RAW && alsa->channels < 9)
     gst_audio_ring_buffer_set_channel_positions (GST_AUDIO_BASE_SINK
@@ -1043,7 +1043,7 @@
       GST_DEBUG_OBJECT (asink, "wait error, %d", err);
     } else {
       GST_DELAY_SINK_LOCK (asink);
-      err = snd_pcm_writei (alsa->handle, ptr, cptr);
+      err = snd_pcm_mmap_writei (alsa->handle, ptr, cptr);
       GST_DELAY_SINK_UNLOCK (asink);
     }
 
